File: SparkCourse/scripts/spark-sql/popular-movies.py
from pyspark.sql import SparkSession
from pyspark.conf import SparkConf
from pyspark.sql.types import StructType, StructField, IntegerType, LongType
from pyspark.sql.functions import col

schema = StructType([
    StructField("customerId", IntegerType(), False),
    StructField("movieId", IntegerType(), False),
    StructField("rating", IntegerType(), False),
    StructField("timeStamp", LongType(), False),
])

spark_session = SparkSession.builder.appName("App").getOrCreate()
df_input = spark_session.read.schema(schema).option("sep", "\t").csv("../../ml-100k/u.data")
popular_movies = df_input.groupBy("movieId").count().orderBy(col("count"))

popular_movies.show(popular_movies.count())


spark_session.stop()

File: SparkCourse/Java/src/main/java/com/learning/spark/dataframes/PopularMovies.java
package com.learning.spark.dataframes;

import com.learning.spark.schemas.Schemas;
import org.apache.spark.SparkConf;
import org.apache.spark.sql.Column;
import org.apache.spark.sql.DataFrameReader;
import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;
import org.apache.spark.sql.SparkSession;
import org.apache.spark.sql.types.StructType;

public class PopularMovies {

    public static void main(String[] args) {
        SparkConf conf = new SparkConf();
        conf.setMaster("local");
        SparkSession spark = SparkSession.builder()
                .config(conf)
                .appName("Java-Popular-Movies").getOrCreate();
        StructType schema = Schemas.Popular_Movies();

        Dataset<Row> df = spark.read()
                .option("sep", "\t")
                .schema(schema)
                .csv("/Volumes/aadi/workspace/mylearnings/SparkCourse/ml-100k/u.data");

        Dataset<Row> sorted = df.groupBy("movieId").count()
                .orderBy("count")
        ;
        sorted.show((int) sorted.count());
        df.printSchema();


    }

}
